### PR TITLE
Fix support bundle

### PR DESCRIPTION
**Pull request by chtyim**
Changes are organized into multiple commits. At a high-level:

* Cleanup / Remove dependencies. k8s dependency shouldn't get bring into CDAP except the cdap-kubernetes module.
* Refactoring / code cleanup to remove warnings, enforce best practices, etc.
* Make the VMInformation available in all services by default.

### CODE REVIEW

**Review by sechegaray — APPROVED** [review]

**Review by sechegaray — COMMENTED** [review]
There are some errors in the build on the messaging module.

File: cdap-support-bundle/src/main/java/io/cdap/cdap/support/internal/app/services/SupportBundleService.java [hunk 1]
```diff
@@ -111,21 +107,27 @@ public String generateSupportBundle(SupportBundleConfiguration supportBundleConf
       .build();
     addToStatus(supportBundleStatus, uuidPath.getPath());
 
-    SupportBundleJob supportBundleJob =
-      new SupportBundleJob(supportBundleTaskFactories, executorService, cConf, supportBundleStatus);
-    SupportBundleTaskConfiguration supportBundleTaskConfiguration =
-      new SupportBundleTaskConfiguration(supportBundleConfiguration, uuid, uuidPath, namespaces, supportBundleJob);
-
+    ExecutorService executor = Executors.newFixedThreadPool(cConf.getInt(Constants.SupportBundle.MAX_THREADS),
```
Comment (dli357) [inline cdap-support-bundle/src/main/java/io/cdap/cdap/support/internal/app/services/SupportBundleService.java]: Are we creating a new executor service every time somebody calls the generate support bundle endpoint? Shouldn't the executor pool be created during startup?

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java [hunk 2]
```diff
@@ -319,9 +321,13 @@ protected void configure() {
         protected void configure() {
           bind(AccessControllerInstantiator.class).toInstance(accessControllerInstantiator);
```
Comment (dli357) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java:322]: Do we need to bind the AccessControllerInstantiator?

File: cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java [hunk 3]
```diff
@@ -71,12 +72,12 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
                                     String programName, @Nullable @QueryParam("run") String run,
                                   @Nullable @QueryParam("maxRunsPerProgram") @DefaultValue("1")
                                     Integer maxRunsPerProgram) throws Exception {
-    /** ensure the user has authentication to create supportBundle */
+    // ensure the user has authentication to create supportBundle
```
Comment (dli357) [inline cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java]: nit: "user is authorized to create..."

File: cdap-support-bundle/src/main/java/io/cdap/cdap/support/internal/app/services/SupportBundleService.java (hunk 1 above)
Comment (dli357) [reply cdap-support-bundle/src/main/java/io/cdap/cdap/support/internal/app/services/SupportBundleService.java]: Ah please disregard, this is fixed in the most recent push.

**Review by dli357 — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java (hunk 2 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java:322]: That's existing code, I assume the binding is correct

**Review by sechegaray — APPROVED** [review]

**Pull request merged**